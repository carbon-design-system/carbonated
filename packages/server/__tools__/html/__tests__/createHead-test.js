/**
 * @jest-environment node
 */

'use strict';

describe('createHead', () => {
  let assetManifest;
  let mockRuntime;
  let createHead;

  beforeEach(() => {
    mockRuntime = 'var runtime = true;';
    assetManifest = require('../__fixtures__/asset-manifest.json');
    createHead = require('../createHead');
  });

  it('should generate the head for the html response', () => {
    expect(createHead(assetManifest, mockRuntime, '', 'App')).toMatchSnapshot();
  });

  it('should embed the runtime', () => {
    const head = createHead(assetManifest, mockRuntime, '', 'App');

    expect(head).toEqual(
      expect.stringContaining(`<script>${mockRuntime}</script>`)
    );
  });

  it('should generate a <link> element for the main.css file', () => {
    const head = createHead(assetManifest, mockRuntime, '', 'App');
    expect(head).toEqual(
      // eslint-disable-next-line quotes
      expect.stringContaining(
        `<link rel="stylesheet" href="${assetManifest['main.css']}"`
      )
    );
  });
});
