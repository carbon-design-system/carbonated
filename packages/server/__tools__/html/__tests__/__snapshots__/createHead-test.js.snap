// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createHead should generate the head for the html response 1`] = `"<head><meta charset=\\"utf-8\\"><title>App</title><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, shrink-to-fit=no\\"><link rel=\\"manifest\\" href=\\"/static/manifest.json?v1.0\\"><link rel=\\"shortcut icon\\" href=\\"/static/favicon.ico?v1.0\\"><link rel=\\"preload\\" href=\\"static/css/main.1c0d3fed.css\\" as=\\"style\\"><link rel=\\"preload\\" href=\\"static/js/main.85d0f27c.js\\" as=\\"script\\"><link rel=\\"preload\\" href=\\"static/js/vendor.7f3d1409.js\\" as=\\"script\\"><link rel=\\"prefetch\\" href=\\"/static/js/Foo.d32c4417.chunk.js\\" as=\\"script\\"><link rel=\\"stylesheet\\" href=\\"static/css/main.1c0d3fed.css\\"><script>var runtime = true;</script></head>"`;
